libethereum/State.cpp

```diff
@@ -512,7 +512,10 @@ void State::rollback(size_t _savepoint)
 			m_cache.erase(change.address);
 			break;
 		case Change::NewCode:
-			account.resetCode();
+			if (change.oldCode.empty())
```

Comment:
Here I would go with `account.setNewCode(std::move(change.oldCode));` all the time. Moving empty bytes is not more costly than reset.